packages/data-context/src/actions/AppActions.ts

```diff
@@ -1,24 +1,43 @@
+import type Bluebird from 'bluebird'
 import type { FoundBrowser } from '@packages/types'
 import pDefer from 'p-defer'
 
 import type { DataContext } from '..'
 
 export interface AppApiShape {
   getBrowsers(): Promise<FoundBrowser[]>
+  ensureAndGetByNameOrPath(nameOrPath: string, returnAll?: boolean, browsers?: FoundBrowser[]): Bluebird<FoundBrowser | FoundBrowser[] | undefined>
```

Comment:
I think we can just type this as `Promise<......>`. As we move forward I think we should just standardize on native promises (with which Bluebird should be compatible, anyway).

Reply:
It needs to be Bluebird since certain calling code uses Bluebird-specific APIs, like [here](https://github.com/cypress-io/cypress/blob/eeac2a28b572b5fbd99b5e295f341e04b73571b1/packages/server/lib/modes/run.js#L1557).